### DISCUSSION TITLE
Can I connect a private network to a set of public machines

### DISCUSSION THREAD

**Discussion by MikeDevresse** [gravitl/netmaker#1880] [Q&A]
Hi everyone,
For a project I want to have a staging environment that runs on virtual machine that has no public IP, the goal of this cluster is to be able to test without letting anyone from outside of my network access it. I also want to have a production environment that runs on VPS. 
The issue is that I want to have a Rancher and an ArgoCD connected to both of the clusters and I want them in my private network to only allow access from within the network. The problem is that the clusters needs to communicate with rancher and if it is in a closed environment, it normally can't access it.
I had the idea to deploy Netmaker to create a tunnel to allow the production cluster to communicate with some components of the private one. Can I do this ? If yes where do my NetMaker needs to be ? In the private network or on a VPS ? I am pretty sure I can do this but I just want to confirm.
Thanks for your time

**Comment by eleaner** [comment]
one up
I have similar issue

my internet provider doe not supply public IP - not even dynamic
I have a few servers at home and a bunch of VPS in the cloud
I have setup v0.17.1 netmaker in the cloud; I have a nice mesh in the graphs
DNS is working
but my machines at home are unable to ping the ones in the cloud and the other way round
they can ping each other in their respective "locations."
netmaker doe not report any errors; all nodes are blue

Obviously, the VPSes cannot directly see my machines at home, but I was expecting the tunnel to be started from the other end.

**Reply by MikeDevresse** [reply]
I've kinda solved my issue by creating a network and in order to protect my more sensible nodes I've added firewalls rules to my netmaker network.
Concerning your issue I think you can solve it by making your netmaker a relay

**Reply by eleaner** [reply]
thanks, 
I thought that too, but it works even less. The relayed nodes just stop being able to communicate - even with themselves.
I filed a bug.
